packages/fx-core/src/plugins/resource/identity/index.ts

```diff
@@ -65,6 +67,17 @@ export class IdentityPlugin implements Plugin {
     );
     this.config.location = ContextUtils.getConfigString(Constants.solution, Constants.location);
 
+    try {
+      ctx.logProvider?.info(Message.checkProvider);
+      const credentials = await ctx.azureAccountProvider!.getAccountCredentialAsync();
+      const resourceManagementClient = new Providers(
+        new ResourceManagementClientContext(credentials!, this.config.azureSubscriptionId)
+      );
+      await resourceManagementClient.register(Constants.resourceProvider);
```

Comment:
Just realized we will try to register resource provider every time? If user does not have permission to register resource provider, but the resource provider is already registered, will this operation fail?

Reply:
We try to register the resource every time. If the register failed, we continue to provide the resource. So the permission to register resource will not interrupt the provision.